File: 2024/Python/Challenge#11.py

def decode_filename(filename: str) -> str:
    filename = "_".join(filename.split("_")[1:])
    filename = ".".join(filename.split(".")[:-1])
    return filename


filename = decode_filename('2023122512345678_sleighDesign.png.grinchwa')
print(filename)  # ➞ "sleighDesign.png"

filename2 = decode_filename('42_chimney_dimensions.pdf.hack2023')
print(filename2)  # ➞ "chimney_dimensions.pdf"

filename3 = decode_filename('987654321_elf-roster.csv.tempfile')
print(filename3)  # ➞ "elf-roster.csv"

File: 2024/JavaScript/Challenge#11.js

/**
 * @param {string} filename - The filename to decode.
 * @returns {string} The decoded filename.
 */
function decodeFilename(filename) {
    filename = filename.split('_').slice(1).join('_');
    filename = filename.split('.').slice(0, -1).join('.');
    return filename;
}


const filename = decodeFilename('2023122512345678_sleighDesign.png.grinchwa')
console.log(filename)  // ➞ "sleighDesign.png"

const filename2 = decodeFilename('42_chimney_dimensions.pdf.hack2023')
console.log(filename2)  // ➞ "chimney_dimensions.pdf"

const filename3 = decodeFilename('987654321_elf-roster.csv.tempfile')
console.log(filename3)  // ➞ "elf-roster.csv"
